tensorflow-utils/src/main/java/org/tensorflow/util/buffer/DataBuffers.java

```diff
@@ -0,0 +1,509 @@
+ */
+package org.tensorflow.util.buffer;
+
+import java.nio.ByteBuffer;
+import java.nio.DoubleBuffer;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+import java.nio.LongBuffer;
+import java.nio.ShortBuffer;
+import org.tensorflow.util.buffer.adapter.BooleanDataAdapter;
+import org.tensorflow.util.buffer.adapter.DataAdapter;
+import org.tensorflow.util.buffer.adapter.DoubleDataAdapter;
+import org.tensorflow.util.buffer.adapter.FloatDataAdapter;
+import org.tensorflow.util.buffer.adapter.IntDataAdapter;
+import org.tensorflow.util.buffer.adapter.LongDataAdapter;
+import org.tensorflow.util.buffer.adapter.ShortDataAdapter;
+import org.tensorflow.util.buffer.impl.jdk.ByteJdkDataBuffer;
+import org.tensorflow.util.buffer.impl.jdk.DoubleJdkDataBuffer;
+import org.tensorflow.util.buffer.impl.jdk.FloatJdkDataBuffer;
+import org.tensorflow.util.buffer.impl.jdk.IntJdkDataBuffer;
+import org.tensorflow.util.buffer.impl.jdk.LongJdkDataBuffer;
+import org.tensorflow.util.buffer.impl.jdk.ShortJdkDataBuffer;
+import org.tensorflow.util.buffer.impl.misc.ArrayDataBuffer;
+import org.tensorflow.util.buffer.impl.misc.BitSetDataBuffer;
+import org.tensorflow.util.buffer.impl.misc.BooleanArrayDataBuffer;
+import org.tensorflow.util.buffer.impl.raw.BooleanRawDataBuffer;
+import org.tensorflow.util.buffer.impl.raw.ByteRawDataBuffer;
+import org.tensorflow.util.buffer.impl.raw.DoubleRawDataBuffer;
+import org.tensorflow.util.buffer.impl.raw.FloatRawDataBuffer;
+import org.tensorflow.util.buffer.impl.raw.IntRawDataBuffer;
+import org.tensorflow.util.buffer.impl.raw.LongRawDataBuffer;
+import org.tensorflow.util.buffer.impl.raw.ShortRawDataBuffer;
+import org.tensorflow.util.buffer.impl.raw.UnsafeReference;
+import org.tensorflow.util.buffer.impl.virtual.BooleanVirtualDataBuffer;
+import org.tensorflow.util.buffer.impl.virtual.DoubleVirtualDataBuffer;
+import org.tensorflow.util.buffer.impl.virtual.FloatVirtualDataBuffer;
+import org.tensorflow.util.buffer.impl.virtual.IntVirtualDataBuffer;
+import org.tensorflow.util.buffer.impl.virtual.LongVirtualDataBuffer;
+import org.tensorflow.util.buffer.impl.virtual.ShortVirtualDataBuffer;
+import org.tensorflow.util.buffer.impl.virtual.VirtualDataBuffer;
+
+/**
+ * Helper class for creating {@code DataBuffer} instances.
+ */
+public final class DataBuffers {
+
+  /**
+   * Creates a buffer of bytes that can store up to {@code size} values
+   *
+   * @param size size of the buffer to allocate
+   * @return a new buffer
+   */
+  public static ByteDataBuffer ofBytes(long size) {
+    return UnsafeReference.isAvailableInternally() ? ByteRawDataBuffer.allocate(size) : ByteJdkDataBuffer.allocate(size);
+  }
+
+  /**
+   * Wraps an array of bytes into a data buffer.
+   *
+   * @param array array to wrap
+   * @param readOnly true if the buffer created must be read-only
+   * @return a new buffer
+   */
+  public static ByteDataBuffer wrap(byte[] array, boolean readOnly) {
```

Comment:
These methods should note if they copy or present a view, as views can be modified through other access points (e.g. the input array here). That's likely to cause odd bugs, so we need to document it so we can point at the documentation when people run into the bugs. It might also be worth adding a `boolean makeCopy` argument to all these methods to make it explicit.

Reply:
Yes, Javadoc needs to be improved here as well. I'll check for the `makeCopy` as well, sounds like a good addition.